frontend/components/Corpora.js

```diff
@@ -0,0 +1,148 @@
+import React, {useEffect, useState} from "react";
+// import STYLES from "./Corpora.module.scss";
+import {getCookie} from "../common";
+import {Modal} from "react-bootstrap";
+
+const Corpora = () => {
+    const [corporaData, setCorporaData] = useState([]);
+    const [newCorpusData, setNewCorpusData] = useState({
+        "title": "",
+        "description": ""
+    });
+    const [loading, setLoading] = useState(true);
+    const [addingCorpus, setAddingCorpus] = useState(false);
+    const [showModal, setShowModal] = useState(false);
+    const handleShowModal = () => setShowModal(true);
+    const handleCloseModal = () => setShowModal(false);
+
+    useEffect(() => {
+        fetch("/api/all_corpora")
+            .then(response => response.json())
+            .then(data => {
+                setCorporaData(data);
+                setLoading(false);
+            });
+    }, []);
+
+    const handleTitleInputChange = (event) => {
+        setNewCorpusData((values) => ({
+            ...values,
+            title: event.target.value
+        }));
+    };
+
+    const handleDescriptionInputChange = (event) => {
+        setNewCorpusData((values) => ({
+            ...values,
+            description: event.target.value
+        }));
+    };
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        setAddingCorpus(true);
+        handleCloseModal();
+        const csrftoken = getCookie("csrftoken");
+        const requestOptions = {
+            method: "POST",
+            header: {
```

Comment:
We talked about this in person, but just leaving a note here: this property name should be `headers`.